src/ssl.c

```diff
@@ -45875,14 +46577,68 @@ int wolfSSL_BN_set_bit(WOLFSSL_BIGNUM* bn, int n)
     }
 
     if (mp_set_bit((mp_int*)bn->internal, n) != MP_OKAY) {
-        WOLFSSL_MSG("mp_set_int error");
+        WOLFSSL_MSG("mp_set_bit error");
         return WOLFSSL_FAILURE;
     }
 
     return WOLFSSL_SUCCESS;
 }
 
 
+int wolfSSL_BN_clear_bit(WOLFSSL_BIGNUM* bn, int n)
+{
+    int ret = WOLFSSL_FAILURE;
+#ifndef WOLFSSL_SMALL_STACK
+    mp_int res[1];
+    mp_int tmp[1];
+#else
+    mp_int* res = null;
+    mp_int* tmp = null;
+#endif
+
+    if (bn == NULL || bn->internal == NULL) {
+        WOLFSSL_MSG("bn NULL error");
+        goto cleanup;
+    }
+    if (mp_is_bit_set((mp_int*)bn->internal, n)) {
+#ifdef WOLFSSL_SMALL_STACK
+       res = (mp_int*)XMALLOC(sizeof(mp_int), NULL, DYNAMIC_TYPE_BIGINT);
+       if (res == NULL) {
+           goto cleanup;
+       }
+       tmp = (mp_int*)XMALLOC(sizeof(mp_int), NULL, DYNAMIC_TYPE_BIGINT);
+       if (tmp == NULL) {
+           goto cleanup;
+       }
+#endif
+        if (mp_init(tmp) != MP_OKAY) {
+            goto cleanup;
+        }
+        if (mp_init(res) != MP_OKAY) {
+            goto cleanup;
+        }
+        if (mp_set_bit(tmp, n) != MP_OKAY) {
+            goto cleanup;
+        }
+        if (mp_sub((mp_int*)bn->internal, tmp, res) != MP_OKAY) {
```

Comment:
Why use res?

Reply:
Can parameters `b == c` in mp_sub? If not then it is necessary to have another variable to store the results.

Reply:
Can `b == c` in mp_sub? If these parameters cannot be the same pointer then it is necessary to use `res`.

Reply:
I would assume that a == c or b == c will work!
Test it out.

Reply:
Looks like `a == c` works.